### PR TITLE
modified renderButtons() in drawer/Menu.jsx (issue 1526)

### PR DESCRIPTION
**Pull request by diegosanmartino**
config's ex:
"name": "Measure",
                "override": {
                    "DrawerMenu": {
                        "buttonConfig": {
                          **"tooltip": "Measure"**,
                            "buttonClassName": "square-button no-border"
                        },
                        "glyph": "1-stilo",
                        "position": 4
                    }
                }

### CODE REVIEW

File: web/client/plugins/drawer/Menu.jsx [hunk 1]
```diff
@@ -58,11 +58,28 @@ var Menu = React.createClass({
         );
     },
     renderButtons() {
-        return this.props.children.map((child) => (
-            <Button key={child.props.eventKey} bsSize="large" className={(child.props.buttonConfig && child.props.buttonConfig.buttonClassName) ? child.props.buttonConfig.buttonClassName : "square-button"} onClick={this.props.onChoose.bind(null, child.props.eventKey)} bsStyle={this.props.activeKey === child.props.eventKey ? 'default' : 'primary'}>
-                {child.props.glyph ? <Glyphicon glyph={child.props.glyph} /> : child.props.icon}
-            </Button>
-        ));
+        let toReturn = [];
```
Comment (mbarto) [inline web/client/plugins/drawer/Menu.jsx]: Please, don't do sideeffects, the map function on arrays returns an array, don't use it as a forEach.

Something more similar to this is preferrable:
```
return this.props.children.map((child) => {
    const button = (<Button key={child.props.eventKey} bsSize="large" className={(child.props.buttonConfig && child.props.buttonConfig.buttonClassName) ? child.props.buttonConfig.buttonClassName : "square-button"} onClick={this.props.onChoose.bind(null, child.props.eventKey)} bsStyle={this.props.activeKey === child.props.eventKey ? 'default' : 'primary'}>
                        {child.props.glyph ? <Glyphicon glyph={child.props.glyph} /> : child.props.icon}
                    </Button>);
    const tooltip = <Tooltip key={"tooltip." + child.props.eventKey} id={"tooltip." + child.props.eventKey}>{child.props.tooltip || ''}</Tooltip>;
    return child.props.tooltip ? (
        <OverlayTrigger placement={"top"} key={"overlay-trigger." + child.props.eventKey}
            overlay={tooltip}>
            {button}
        </OverlayTrigger>
    ) : button;
});
```

Comment (diegosanmartino) [inline web/client/plugins/drawer/Menu.jsx]: ok

File: web/client/plugins/drawer/Menu.jsx [hunk 2]
```diff
@@ -6,7 +6,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 var React = require('react');
-var {Glyphicon, Button} = require('react-bootstrap');
+var {Glyphicon, Button, OverlayTrigger, Tooltip} = require('react-bootstrap');
 var Sidebar = require('react-sidebar').default;
```
Comment (mbarto) [inline web/client/plugins/drawer/Menu.jsx]: var Message = require('../../components/I18N/Message');

File: web/client/plugins/drawer/Menu.jsx [hunk 3]
```diff
@@ -58,11 +58,18 @@ var Menu = React.createClass({
...
-                {child.props.glyph ? <Glyphicon glyph={child.props.glyph} /> : child.props.icon}
-            </Button>
-        ));
+        return this.props.children.map((child) => {
+            const button = (<Button key={child.props.eventKey} bsSize="large" className={(child.props.buttonConfig && child.props.buttonConfig.buttonClassName) ? child.props.buttonConfig.buttonClassName : "square-button"} onClick={this.props.onChoose.bind(null, child.props.eventKey)} bsStyle={this.props.activeKey === child.props.eventKey ? 'default' : 'primary'}>
+                        {child.props.glyph ? <Glyphicon glyph={child.props.glyph} /> : child.props.icon}
+                    </Button>);
+            const tooltip = <Tooltip key={"tooltip." + child.props.eventKey} id={"tooltip." + child.props.eventKey}>{child.props.buttonConfig.tooltip || ''}</Tooltip>;
```
Comment (mbarto) [inline web/client/plugins/drawer/Menu.jsx]:
```
const tooltip = <Tooltip key={"tooltip." + child.props.eventKey} id={"tooltip." + child.props.eventKey}><Message msgId={child.props.buttonConfig.tooltip}/></Tooltip>;
```

File: web/client/plugins/drawer/Menu.jsx [hunk 4]
```diff
@@ -58,11 +58,18 @@ var Menu = React.createClass({
...
+            return child.props.buttonConfig.tooltip ? (
+                <OverlayTrigger placement={"top"} key={"overlay-trigger." + child.props.eventKey}
```
Comment (mbarto) [inline web/client/plugins/drawer/Menu.jsx]: bottom, if you put it at top, it's not visible (did you test it?)

Comment (diegosanmartino) [inline web/client/plugins/drawer/Menu.jsx]: const tooltip = <Tooltip key={"tooltip." + child.props.eventKey} id={"tooltip." + child.props.eventKey}><Message msgId={child.props.buttonConfig.tooltip}/></Tooltip>
child.props.buttonConfig.tooltip is correct but the tooltip is empty... 
renderButtons() {
        return this.props.children.map((child) => {
            const button = (<Button key={child.props.eventKey} bsSize="large" className={(child.props.buttonConfig && child.props.buttonConfig.buttonClassName) ? child.props.buttonConfig.buttonClassName : "square-button"} onClick={this.props.onChoose.bind(null, child.props.eventKey)} bsStyle={this.props.activeKey === child.props.eventKey ? 'default' : 'primary'}>
                        {child.props.glyph ? <Glyphicon glyph={child.props.glyph} /> : child.props.icon}
                    </Button>);
            const tooltip = <Tooltip key={"tooltip." + child.props.eventKey} id={"tooltip." + child.props.eventKey}><Message msgId={child.props.buttonConfig.tooltip}/></Tooltip>;
            return child.props.buttonConfig.tooltip ? (
                <OverlayTrigger placement={"bottom"} key={"overlay-trigger." + child.props.eventKey}
                    overlay={tooltip}>
                    {button}
                </OverlayTrigger>
            ) : button;
        });
    },

Comment (mbarto) [inline web/client/plugins/drawer/Menu.jsx]: Do you have the string in your translations file?

Comment (diegosanmartino) [inline web/client/plugins/drawer/Menu.jsx]: ok

**Comment by MV88** [comment]
Hi @diegosanmartino I would suggest you to use the Atom editor (or Sublime) with the plugins suggested here https://geosolutions-it.github.io/MapStore2/developer-guide/building-and-developing in order to see in the editor some syntax error. This would avoid to push commit that will fail the travis build. Thank you

**Pull request merged**